worker/uniter/runner/context/context.go

```diff
@@ -777,6 +777,17 @@ func (ctx *HookContext) Prepare() error {
 	return nil
 }
 
+// Reset implements the Context interface.
+func (ctx *HookContext) Reset() error {
```

Comment:
Context has generic methods like Prepare() and Flush().
Plus task specific ones like ResetExecutionSetUnitStatus()

I think here RequeueAction() is better. We already have an ActionData() method. ie action related methods have "Action" in the name